File: Back/InsurancesAPI/Models/Business/CustomerInsurance.cs
﻿using Models.Enums;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Models.Business
{
    public class CustomerInsurance
    {
        [Key]
        public long CustomerInsuranceID { get; set; }
        public DateTime InitDate { get; set; }
        public decimal Price { get; set; }
        public int MonthsDuration { get; set; }
        public RiskTypeEnum RiskType { get; set; }
        [ForeignKey("Customer")]
        public long CustomerCode { get; set; }
        public Customer Customer { get; set; }
        [ForeignKey("Insurance")]
        public long InsuranceCode { get; set; }
        public Insurance Insurance { get; set; }

        public bool ValidateCustomerInsurance() {
            if (Insurance != null)
            {
                return !((RiskType == RiskTypeEnum.HIGH || RiskType == RiskTypeEnum.MEDIUM_HIGH) && Insurance.Coverage > 50 );
            }
            return false;
        }
        
    }
}

File: Front/GAPInsurancesApp/src/app/models/customerinsurance.ts
import { RiskTypeEnum } from '../framework/enum/insurance-enum';
import { InsuranceModel } from './insurance';
import { CustomerModel } from './customer';

export class CustomerInsuranceModel {
    CustomerInsuranceID: number;
    CustomerCode: number;
    InsuranceCode: number;
    InitDate: Date;
    Price: number;
    MonthsDuration: number;
    RiskType: RiskTypeEnum;

    Customer: CustomerModel;
    Insurance: InsuranceModel;

    CustomerInsuranceModel() {
        this.CustomerInsuranceID = -1;
        this.CustomerCode = -1;
        this.InsuranceCode = -1;
        this.InitDate = new Date();
        this.Price = -1;
        this.MonthsDuration = -1;
        this.RiskType = RiskTypeEnum.None;
    }
}

